File: problems_026to050/problem_050/solution.py
import os
import sys
from math import log

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch
from toolbox import first_primes, is_prime_naive


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = None

        limit = 1000000
        n = 1

        while True:
            approximate_value = n**2 * log(n) / 2

            if approximate_value >= limit:
                break

            n += 1

        primes = first_primes(n)

        for length in range(n, 0, -1):
            for start_index in range(0, n-length+1):
                end_index = start_index + length
                consecutive_sum = sum(primes[start_index:end_index])

                if consecutive_sum < limit and is_prime_naive(consecutive_sum):
                    answer = consecutive_sum
                    break

            if answer is not None:
                break

        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_026to050/problem_050/solution.js
const { sum, isPrime } = require('mathjs');

const solution = () => {
  let answer = null;

  const limit = 1000000;
  let n = 0;
  let approximateValue = 0;

  while (approximateValue < limit) {
    n += 1;
    approximateValue = ((n ** 2) * Math.log(n)) / 2;
  }

  const primes = [2];
  let num = 3;

  while (primes.length < n) {
    if (isPrime(num)) {
      primes.push(num);
    }

    num += 2;
  }

  for (let length = n; length > 0; length -= 1) {
    for (let startIndex = 0; startIndex <= n - length; startIndex += 1) {
      const endIndex = startIndex + length;
      const consecutiveSum = sum(primes.slice(startIndex, endIndex));

      if (consecutiveSum < limit && isPrime(consecutiveSum)) {
        answer = consecutiveSum;
        break;
      }
    }

    if (answer !== null) {
      break;
    }
  }

  return answer;
};

if (process.env.NODE_ENV !== 'test') {
  // eslint-disable-next-line no-console
  console.log(solution());
}

module.exports = solution;
